mongofiles/mongofiles.go

```diff
@@ -216,40 +234,52 @@ func (mf *MongoFiles) findGFSFiles(query bson.M) (files []*gfsFile, err error) {
 }
 
 // Gets the GridFS file the options specify. Use this for the get family of commands.
-func (mf *MongoFiles) getTargetGFSFile() (*gfsFile, error) {
+func (mf *MongoFiles) getTargetGFSFile() ([]*gfsFile, error) {
 	var gridFiles []*gfsFile
 	var err error
 
-	var queryProp string
-	var query string
+	// If mongofiles --get ... is called, then query for all files
+	// specified in mf.FileNameList -- otherwise, preserve correct
+	// behavior for mongofiles --get_id ...
+	if len(mf.FileNameList) > 0 {
+		query := bson.M{"filename": bson.M{"$in": mf.FileNameList}}
```

Comment:
Interesting -- I didn't anticipate this.

Off the top of my head: How would you feel about an equivalent query specifying the exact filename with an [_$or_ operator](https://docs.mongodb.com/manual/reference/operator/query/or/#op._S_or)? Since, in this case, we're specifying the exact file name, I think it should solve the problem you're specifying. Would you be amenable to this?

Reply:
@tfogo I was actually [incorrectly] thinking the same thing. I think -- off the top of my head -- the difference between $in and $regex is that [$regex offers some `<options>` that $in might not](https://docs.mongodb.com/manual/reference/operator/query/regex/#op._S_regex). Case in point, with the `/<pattern>/<options>/` syntax provided by $regex, you would be able extend `--get_regex` to a specific kind of MQL query at the user level.

This is purely just conjecture -- in any case, I'm still "researching" the `--get_regex` ticket -- so it's unclear whether or not that's how we should implement `--get_regex`. With that said, it's worth noting that, if I understand this correctly, there's some options that $regex gives you that $in does not.

Reply:
I agree with @tfogo , if `$in` could just support regex, we can unify the two subcommands, `--get` and `--getregex` to provide support for multiple files as well as regex.

Reply:
If it's only `x` and `s` that can't be used with `$in`, I think it's probably fine to not have those. I suspect that `i` is the only option which users might want to set for getting gridfs files.

Reply:
I'm fine with unifying the two commands, in light of this new detail, as long as you both and @rychipman are all OK with it. However, is it OK that we don't support the `i` option?

Reply:
`i` and `m` are supported with regex literals in `$in`. Looks like only `x` and `s` need `$regex`.

Reply:
Resolved in #286 -- turns out the client cannot use `mongofiles get ...` to query with regex under the current implementation.